lib/glimesh/chat/effects.ex

```diff
@@ -101,31 +110,42 @@ defmodule Glimesh.Chat.Effects do
     [render_avatar(user), " ", render_username(user)]
   end
 
+  @doc """
+  Renders channel badges based on message metadata
+  """
+  def render_channel_badge(%Message{metadata: %Metadata{streamer: true}}) do
+    StreamerBadge.render()
+  end
+
+  def render_channel_badge(%Message{metadata: %Metadata{subscriber: true, moderator: true}}) do
+    [ModeratorBadge.render(), " ", ChannelSubscriberBadge.render()]
+  end
+
+  def render_channel_badge(%Message{metadata: %Metadata{moderator: true}}) do
+    ModeratorBadge.render()
+  end
+
+  def render_channel_badge(%Message{metadata: %Metadata{subscriber: true}}) do
+    ChannelSubscriberBadge.render()
+  end
+
+  def render_channel_badge(%Message{metadata: %Metadata{admin: true}}), do: ""
+  def render_channel_badge(%Message{metadata: nil}), do: nil
+  def render_channel_badge(%Message{metadata: %Metadata{} = _metadata}), do: nil
+
   def render_channel_badge(channel, user) do
```

Comment:
Why do we still keep this function around? Shouldn't the above take care of all of the cases?

Reply:
Assuming I'm not writing a migration here, it was to handle falling back to the old behaviour if required.
Since your preference is to write a migration --- can definitely remove this.